W-16716183 MuleSoft AI Chain Connector (#2706)
mulesoft/docs-connectors: mulesoft-ai-chain/1.0/modules/ROOT/pages/mulesoft-ai-chain-connector-studio.adoc

> = Using Anypoint Studio to Configure MuleSoft Chain AI Connector 1.0.0 - Mule 4

Reviewer: ```suggestion
= Using Anypoint Studio to Configure MuleSoft Chain AI Connector 1.0 - Mule 4
```

include::connectors::partial$t-studio-partial.adoc[tags=topic-intro]

> 

Reviewer: need to define the {connector-name}, {exchange-search-name}, and {studio-connector-name} variables in this .adoc to render the doc properly

include::connectors::partial$t-studio-partial.adoc[tags=property-placeholders]

To configure the global element for MuleSoft AI Chain Connector, follow these steps:

. Select the operation in the Studio canvas.
. On the properties screen for the operation, click the Add (*+*) icon  to access the global element configuration fields.
. In the *Global Elements Properties > General* tab, choose your preferred large language model (LLM) type from *LLM*:
+
** Anthropic
** Azure OpenAI
** Mistral AI
** Ollama
** OpenAI
** GroqAI
. From *Config type*, choose from these configuration types: 
* *Environment Variables*
+
This configuration requires you to set the environment variables in the operating system where Mule runtime is deployed. When you choose this option, enter a `-` in the *File path* field.
+
Based on the LLM type you chose, you must set different environment variables. These are the supported environment variables based on the supported LLM types:
+

> * `# Anthropic: - ANTHROPIC_API_KEY` 

Reviewer: ```suggestion
** Anthropic: `ANTHROPIC_API_KEY` 
```

> * `# AWS Bedrock:- AWS_ACCESS_KEY_ID- AWS_SECRET_ACCESS_KEY` 

Reviewer: ```suggestion
** AWS Bedrock: `AWS_ACCESS_KEY_ID- AWS_SECRET_ACCESS_KEY` 
```

> * `# Azure OpenAI:- AZURE_OPENAI_ENDPOINT- AZURE_OPENAI_KEY- AZURE_OPENAI_DEPLOYMENT_NAME` 

Reviewer: ```suggestion
** Azure OpenAI: `AZURE_OPENAI_ENDPOINT- AZURE_OPENAI_KEY- AZURE_OPENAI_DEPLOYMENT_NAME` 
```

> * `# MistralAI:- MISTRAL_AI_API_KEY` 

Reviewer: ```suggestion
** MistralAI: `MISTRAL_AI_API_KEY` 
```

> * `# OpenAI:- OPENAI_API_KEY` 

Reviewer: ```suggestion
** OpenAI: `OPENAI_API_KEY` 
```

> * `# Ollama:- OLLAMA_BASE_URL` 

Reviewer: ```suggestion
** Ollama: `OLLAMA_BASE_URL` 
```

> * `# Groq AI:- GROQ_API_KEY`

Reviewer: ```suggestion
** Groq AI: `GROQ_API_KEY`
```

* *Configuration JSON* 
+
This configuration requires you to provide a configuration JSON file with all the required LLM properties.
+
This is an example of the configuration JSON file:
+
[source,json]
----
{
    "OPENAI": {
        "OPENAI_API_KEY": "YOUR_OPENAI_API_KEY"
    },
    "MISTRAL_AI": {
        "MISTRAL_AI_API_KEY": "YOUR_MISTRAL_AI_API_KEY"
    },
    "OLLAMA": {
        "OLLAMA_BASE_URL": "http://baseurl.ollama.com"
    },
    "GROQAI_OPENAI": {
        "GROQ_API_KEY": "YOUR_GROQAI_APIKEY"
    },
    "ANTHROPIC": {
        "ANTHROPIC_API_KEY": "YOUR_ANTHROPIC_API_KEY"
    },
    "AZURE_OPENAI": {
        "AZURE_OPENAI_KEY": "YOUR_AZURE_OPENAI_KEY",
        "AZURE_OPENAI_ENDPOINT": "http://endpoint.azure.com",
        "AZURE_OPENAI_DEPLOYMENT_NAME": "YOUR_DEPLOYMENT_NAME"
    }
}
----
+
Make sure you fill out the required properties for your LLM type. You can store the file externally or add it directly in the Mule application under the `src/main/resource` folder by using this Dataweave expression:
+

> `DW Expression: mule.home ++ "/apps/" ++ app.name ++ "/envVars.json"`

Reviewer: ```suggestion
`mule.home ++ "/apps/" ++ app.name ++ "/envVars.json"`
```

> . In *File path*, enter `-` if you chose *Environment Variables* as the configuration type. If you chose *Configuration JSON*, enter the path to the JSON file, or the Dataweave expression if you are storing it in the Mule application under the `src/main/resource` folder.

Reviewer: ```suggestion
. In *File path*, enter `-` if you chose *Environment Variables* as the configuration type. If you chose *Configuration JSON*, enter the path to the JSON file, or the DataWeave expression if you are storing it in the Mule application under the `src/main/resource` folder.
```

. In *Model name*, choose from the supported models for the LLM provider. 

> . For *Temperature* enter a number between `0` and `2`, or use the default value `0.7`. 

Reviewer: ```suggestion
. In *Temperature*, enter a number between `0` and `2`, or use the default value `0.7`. 
```

+
The temperature is used to control the randomness of the output. When you set it higher, you'll get more random outputs. When you set it lower, towards `0`, the values are more deterministic.

> . Enter the number of seconds for the *LLM Timeout* (when to time out the request). The default is `60` seconds. 

Reviewer: ```suggestion
. In *LLM Timeout*, enter the number of seconds of when to time out the request. The default is `60` seconds. 
```

> . *Max tokens* defines the number of LLM Token to use when generating a response. 

Reviewer: ```suggestion
. In *Max tokens*, enter the number of LLM tokens to use when generating a response. 
```

+
This parameter helps control the usage and costs when engaging with LLMs.
. Click *OK*.

> 

Reviewer: missing the Configure the other connector fields header

> 

Reviewer: add [[view-the-app-log]]

> == View the Application Log

Reviewer: ```suggestion
== View the App Log
```

include::connectors::partial$t-studio-partial.adoc[tags=view-app-log]

== See Also

include::connectors::partial$t-studio-partial.adoc[tags=see-also]
* xref:mulesoft-ai-chain-connector-reference.adoc[]

> * xref:mulesoft-ai-chain-additional-config.adoc[]

Reviewer: ```suggestion
* xref:mulesoft-ai-chain-additional-config.adoc[MuleSoft AI Chain - Additional Configuration Information]
```

Reviewer: need to hardcode to avoid mule 4 in the title
